### PR TITLE
[DOC] Update references in nilearn/decomposition with bibtex

### LINKED ISSUE #2782: Use bibtex in nilearn/decomposition
Convert references in:

- `nilearn/decomposition/canica.py`
- `nilearn/decomposition/dict_learning.py`

to use footcite / footbibliography.
See #2759

### PR DESCRIPTION
**Pull request by achamma723**
Closes #2782 
This PR updates the references in ```nilearn/decomposition/canica.py``` and ```nilearn/decomposition/dict_learning.py``` with the bibtex format

### CODE REVIEW

File: nilearn/decomposition/canica.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 @fill_doc
 class CanICA(_MultiPCA):
-    """Perform Canonical Independent Component Analysis [1]_ [2]_.
+    """Perform Canonical Independent Component Analysis :footcite:`Varoquaux2010c` :footcite:`Varoquaux2010d`.
```
Comment (Remi-Gau) [inline nilearn/decomposition/canica.py:21]: CI says this line is too long:
```suggestion
    """Perform Canonical Independent Component Analysis \
    :footcite:`Varoquaux2010c` :footcite:`Varoquaux2010d`.
```
https://github.com/nilearn/nilearn/actions/runs/4314198218/jobs/7526922367#step:5:16

Comment (achamma723) [reply nilearn/decomposition/canica.py:21]: > CI says this line is too long:
> 
> https://github.com/nilearn/nilearn/actions/runs/4314198218/jobs/7526922367#step:5:16

Thank you for reviewing! The fix is committed

File: doc/references.bib [hunk 2]
```diff
@@ -1191,6 +1201,33 @@ @inproceedings{Varoquaux2010b
...
+@article{Varoquaux2010c,
+	title = {A group model for stable multi-subject ICA on fMRI datasets},
+	author = {Varoquaux, Gael
+	and Sadaghiani, Sepideh
+	and Pinel, Philippe
+	and Kleinschmidt, Andreas
+	and Poline, Jean-Baptiste
+	and Thirion, Bertrand},
+	year = {2010},
+    doi={10.1016/j.neuroimage.2010.02.010},
```
Comment (Remi-Gau) [inline doc/references.bib:1213]: CI says that you are missing a journal in those refs:

https://github.com/nilearn/nilearn/actions/runs/4314727598/jobs/7528138270#step:33:609

**Review by bthirion — APPROVED** [review]
LGTM, thx !

File: nilearn/decomposition/canica.py [hunk 3]
```diff
@@ -18,7 +18,8 @@
 
 @fill_doc
 class CanICA(_MultiPCA):
-    """Perform Canonical Independent Component Analysis [1]_ [2]_.
+    """Perform Canonical Independent Component Analysis \
```
Comment (NicolasGensollen) [inline nilearn/decomposition/canica.py:21]:
```suggestion
    """Perform :term:`Canonical Independent Component Analysis<CanICA>`  \
```

**Review by NicolasGensollen — APPROVED** [review]
Thanks @achamma723 !
LGTM with a small suggestion.
I think we were adding entries in the changelog for these bib PRs but I'm not 100% sure. (@ymzayek ?)

**Comment by ymzayek** [comment]
> I think we were adding entries in the changelog for these bib PRs but I'm not 100% sure. (@ymzayek ?)

Yes we have been so let's add one for this one too.

Also there are some error in the [Documentation builder](https://github.com/nilearn/nilearn/actions/runs/4314955030/jobs/7528671816#step:33:665):

```bash
/home/runner/work/nilearn/nilearn/doc/modules/decomposition.rst:19:<autosummary>:1: ERROR: Too many autonumbered footnote references: only 0 corresponding footnotes available.
/home/runner/work/nilearn/nilearn/doc/modules/decomposition.rst:19:<autosummary>:1: ERROR: Unknown target name: "footcite-varoquaux2010c".
/home/runner/work/nilearn/nilearn/doc/modules/decomposition.rst:19:<autosummary>:1: ERROR: Unknown target name: "footcite-varoquaux2010d".
/home/runner/work/nilearn/nilearn/doc/modules/decomposition.rst:19:<autosummary>:1: ERROR: Unknown target name: "footcite-mensch2016".
```

I haven't had time to look into why but met me know if you need help resolving them

File: nilearn/decomposition/canica.py [hunk 4]
```diff
@@ -18,7 +18,8 @@
 
 @fill_doc
 class CanICA(_MultiPCA):
-    """Perform Canonical Independent Component Analysis [1]_ [2]_.
+    """Perform :term:`Canonical Independent Component Analysis<CanICA>`  \
+    :footcite:`Varoquaux2010c` :footcite:`Varoquaux2010d`.
```
Comment (ymzayek) [inline nilearn/decomposition/canica.py:22]:
```suggestion
    """Perform :term:`Canonical Independent Component Analysis<CanICA>`.
    
    See :footcite:`Varoquaux2010c` and :footcite:`Varoquaux2010d`.
```

A quick fix for the errors in https://github.com/nilearn/nilearn/pull/3558#issuecomment-1455720506
Seems to be a problem on Sphinx side. The links in the API of the functions work but render incorrectly in the summary page: 
Screenshot from 2023-03-17 12-29-31

File: nilearn/decomposition/dict_learning.py [hunk 5]
```diff
@@ -37,7 +37,7 @@ def _compute_loadings(components, data):
 @fill_doc
 class DictLearning(_BaseDecomposition):
     """Perform a map learning algorithm based on spatial component sparsity, \
-    over a :term:`CanICA` initialization [1]_.
+    over a :term:`CanICA` initialization :footcite:`Mensch2016`.
```
Comment (ymzayek) [inline nilearn/decomposition/dict_learning.py:40]:
```suggestion
    over a :term:`CanICA` initialization. 
    
    See :footcite:`Mensch2016`.
```

Same as above.

**Review by ymzayek — APPROVED** [review]

**Pull request merged**